### PR TITLE
feat: store partition info in ceresmeta

### PR DESCRIPTION
**Pull request by chunshao90**
# Which issue does this PR close?

Closes #

# Rationale for this change
 In order to support the case where the partition table is opened by multiple machines in a non-cloud environment, store the partition info of `PartitionTable` in `ceresmeta`.
 

# What changes are included in this PR?
* Store partition info of `PartitionInfo` in metadata of the table.

# Are there any user-facing changes?
No.

# How does this change test
Manual testing.

### CODE REVIEW

File: server/cluster/table_manager_test.go [hunk 1]
```diff
@@ -191,7 +191,12 @@ func testCreateTable(ctx context.Context, re *require.Assertions, manager Manage
 ) {
 	c, err := manager.GetCluster(ctx, clusterName)
 	re.NoError(err)
-	table, err := c.CreateTable(ctx, shardID, schema, tableName, false)
+	table, err := c.CreateTable(ctx, CreateTableRequest{
+		ShardID:       shardID,
+		SchemaName:    schema,
+		TableName:     tableName,
+		PartitionInfo: storage.PartitionInfo{},
```
Comment (ZuLiangWang) [inline server/cluster/table_manager_test.go:198]: Remove it, `table_manager_test` used to test normal table, `PartitionInfo` should be nil.
If you want to test create partition table in cluster, you should define the new function`testCreatePartitionTable`.

File: server/coordinator/procedure/create_drop_partition_table_test.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ func TestCreateAndDropPartitionTable(t *testing.T) {
 
 	shardPicker := NewRandomBalancedShardPicker(manager)
 
-	testTableNum := 20
+	testTableNum := 8
```
Comment (ZuLiangWang) [inline server/coordinator/procedure/create_drop_partition_table_test.go:24]: Why change the number of tables?

File: server/cluster/table_manager_test.go (hunk 1 above)
Comment (chunshao90) [reply server/cluster/table_manager_test.go:198]: `storage.PartitionInfo{}` means the table is not a `PartitionTable`.

File: server/coordinator/procedure/create_drop_partition_table_test.go (hunk 2 above)
Comment (chunshao90) [reply server/coordinator/procedure/create_drop_partition_table_test.go:24]: There is no need to create so many partition tables, the current value has already covered the required path.

File: server/storage/types.go [hunk 3]
```diff
@@ -163,12 +163,20 @@ type Schema struct {
...
-	Partitioned bool
+	ID            TableID
+	Name          string
+	SchemaID      SchemaID
+	CreatedAt     uint64
+	PartitionInfo PartitionInfo
+}
+
+func (t Table) IsPartitioned() bool {
+	return t.PartitionInfo.Info != nil
```
Comment (ZuLiangWang) [inline server/storage/types.go:179]: Add check for `t.PartitionInfo`.

**Review by ZuLiangWang — APPROVED** [review]
LGTM

**Pull request merged**